CryptoExample1 (#18)
deephaven/examples: cryptocurrency/README.md

> # cryptocurrency

Reviewer: ```suggestion
# Cryptocurrency
```

> The script has will pull live and historical for specifed cryptocurencies from the [CoinGecko](https://www.coingecko.com/) website into the Deephave Community Edition IDE.

Reviewer: ```suggestion
The script will pull live and historical data for specified cryptocurencies from the [CoinGecko](https://www.coingecko.com/) website into the Deephaven Community Core IDE.
```

> ## Variables ##

Reviewer: ```suggestion
## Variables
```

> - **timeToWatch:** Integer number of minutes to run the script

Reviewer: ```suggestion
- **timeToWatch:** Integer number of minutes to run the script.
```

Reviewer: Shouldn't these be code? `timeToWatch`, `secondsToSleep` etc.?

> - **secondsToSleep:** Integer number of seconds between data pulls.  Note that too frequent of requests will generate a HTTP status '429 Too Many Requests'. Recommend this number of 10 or higher.

Reviewer: ```suggestion
- **secondsToSleep:** Integer number of seconds between data pulls.  Note that too frequent of requests will generate an HTTP status '429 Too Many Requests'. It is recommended that this number does not exceed 10 or higher.
```

> - **getHistory:** Boolean value, if one wants only live data or live and historical

Reviewer: ```suggestion
- **getHistory:** Boolean value, if one wants only live data or live and historical.
```

Reviewer: Should we say `true` for both live and historical data?

> - **daysHistory:** Integer number of days to collect history. Data has automatic granularity. Minutely data will be used for duration within 1 day, Hourly data will be used for duration between 1 day and 90 days, Daily data will be used for duration above 90 days.

Reviewer: ```suggestion
- **daysHistory:** Integer number of days to collect history. Data has automatic granularity. 
   - Minutely data will be used for a duration within 1 day.
   - Hourly data will be used for a duration between 1 day and 90 days.
    - Daily data will be used for a duration above 90 days.
```

Reviewer: Is minutely a typical word?

Author: it is a word. Even looked it up :) https://english.stackexchange.com/questions/3091/weekly-daily-hourly-minutely

> - **ids:** Array of coins to pull information. For a full and current list of values execute the following: curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"

Reviewer: ```suggestion
- **ids:** Array of coins to pull information. For a full and current list of values execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`
```

# Source and License

> This script was build using the CoinGecko crypto API.  This allows one to track over 7,000 coins such as bitcoin, litecoin, ethereum, and more from more than 400 exchanges and growing.

Reviewer: ```suggestion
This script was built using the CoinGecko crypto API.  This allows one to track over 7,000 coins such as bitcoin, litecoin, and ethereum from more than 400 exchanges and growing.
```

# Cryptocurrency

> The script will pull live and historical data for specified cryptocurencies from the [CoinGecko](https://www.coingecko.com/) website into [Deephaven Community Core](****link***).

Reviewer: ```suggestion
The script will pull live and historical data for specified cryptocurencies from the [CoinGecko](https://www.coingecko.com/) website into [Deephaven Community](****link***).
```

## Variables

> - **timeToWatch:** Integer number of minutes to run the script.

Reviewer: Just wondering why these are bold and not in backticks.

> - **`secondsToSleep`:** Integer number of seconds between data pulls.  Note that too frequent of requests will generate an HTTP status '429 Too Many Requests'. It is recommended that this number is 10 or higher.

Reviewer: ```suggestion
- **`secondsToSleep`:** Integer number of seconds between data pulls.  Note that too frequent requests will generate an HTTP status '429 Too Many Requests'. It is recommended that this number is 10 or higher.
```

> - **`getHistory`:** Boolean value, 

Reviewer: ```suggestion
- **`getHistory`:** Boolean value.
```

   -  `false` for only live data.
   -  `true` for live and historical data.
- **daysHistory:** Integer number of days to collect history. Data has automatic granularity. 
   - Minutely data will be used for a duration within 1 day.
   - Hourly data will be used for a duration between 1 day and 90 days.
   - Daily data will be used for a duration above 90 days.

> - **ids:** Array of coins to pull information. For a full and current list of values execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`

Reviewer: ```suggestion
- **ids:** Array of coins used to pull information. For a full and current list of values, execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`
```

# Outcome

> Upon running the script in your Deephaven IDE the `result` table will be created.  

Reviewer: ```suggestion
Upon running the script in your Deephaven IDE, the `result` table will be created.  
```

> This table can be sorted on the DateTime column to see new data streaming in.

Reviewer: ```suggestion
This table can be sorted on the `DateTime` column to see new data streaming in.
```

![img](./crypto1.png)

> 

Reviewer: ```suggestion
The result table includes the following columns:
```

> The result table includes the following columns:

Reviewer: ```suggestion

The result table includes the following columns:

```

> - **DateTime:** The date and time that the coin value was updated.  Changes are queried every `secondsToSleep`.  Only new data will be added to the table.  If the coin value did not change then no new data will populate. 

Reviewer: ```suggestion
- **DateTime:** The date and time that the coin value was updated.  Changes are queried every `secondsToSleep`.  Only new data will be added to the table.  If the coin value did not change, then no new data will populate. 
```

- **Coins:** The name of the coin. 

> - **prices:** refers to the current global volume-weighted average price of a coin traded on an active cryptoasset exchange as tracked by CoinGecko.

Reviewer: ```suggestion
- **prices:** Refers to the current global volume-weighted average price of a coin traded on an active cryptoasset exchange as tracked by CoinGecko.
```

> - **market_caps:** is one of the metrics used to measure the relative size of a coin. Market Capitalization is calculated by multiplying Coin Price with Available Supply.

Reviewer: ```suggestion
- **market_caps:** One of the metrics used to measure the relative size of a coin. Market Capitalization is calculated by multiplying Coin Price with Available Supply.
```

> - **total_volumes:** is the total trading volume of a coin across all active coin exchanges tracked by CoinGecko.

Reviewer: ```suggestion
- **total_volumes:** The total trading volume of a coin across all active coin exchanges tracked by CoinGecko.
```

- **`daysHistory`:** Integer number of days to collect history. Data has automatic granularity. 
   - Minutely data will be returned for a duration within 1 day.
   - Hourly data will be returned for a duration between 1 day and 90 days.
   - Daily data will be returned for a duration above 90 days.
- **`ids`:** Array of coins to pull information on. For a full and current list of values, execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`

# Outcome

Upon running the script in your Deephaven IDE, the `result` table will be created.  

This table can be sorted on the `Timestamp` column to see new data streaming in.

![img](./crypto1.png)

The result table includes the following columns:

- **`Timestamp`:** The date and time that the coin value was updated.
- **`Coin`:** The name of the coin.
- **`Price`:** Refers to the current global volume-weighted average price of a coin traded on an active cryptoasset exchange as tracked by CoinGecko.
- **`MarketCap`:** One of the metrics used to measure the relative size of a coin. Market Capitalization is calculated by multiplying Coin Price with Available Supply.

> - **`total_volume`:** The total trading volume of a coin across all active coin exchanges tracked by CoinGecko.

Reviewer: ```suggestion
- **`TotalVolume`:** The total trading volume of a coin across all active coin exchanges tracked by CoinGecko.
```

## Variables
- **`timeToWatch`:** Integer number of minutes to run the script.
- **`secondsToSleep`:** Integer number of seconds between data pulls.  It is recommended that this number is 10 or higher. Note that too frequent requests will generate an HTTP status: '429 Too Many Requests'. 
- **`getHistory`:** Boolean value.
   -  `false` for only live data.
   -  `true` for live and historical data.
- **`daysHistory`:** Integer number of days to collect history. Data has automatic granularity. 
   - Minutely data will be used for a duration within 1 day.
   - Hourly data will be used for a duration between 1 day and 90 days.
   - Daily data will be used for a duration above 90 days.
- **`ids`:** Array of coins used to pull information. For a full and current list of values, execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`

# Outcome

Upon running the script in your Deephaven IDE, the `result` table will be created.  

This table can be sorted on the `DateTime` column to see new data streaming in.

![img](./crypto1.png)

The result table includes the following columns:

> - **`DateTime`:** The date and time that the coin value was updated.  Changes are queried every `secondsToSleep`.  Only new data will be added to the table.  If the coin value did not change, then no new data will populate. 

Reviewer 2: I would consider removing the last two or three sentences, since they are more about how the program works and not about the column.

- **`Coin`:** The name of the coin. 
- **`price`:** Refers to the current global volume-weighted average price of a coin traded on an active cryptoasset exchange as tracked by CoinGecko.
- **`market_cap`:** One of the metrics used to measure the relative size of a coin. Market Capitalization is calculated by multiplying Coin Price with Available Supply.

> - **`total_volume`:** The total trading volume of a coin across all active coin exchanges tracked by CoinGecko.

Reviewer 2: As a convention we use camel case for column names.  e.g. `MarketCap` and `TotalVolume` and `Price`.

# Cryptocurrency

The script will pull live and historical data for specified cryptocurencies from the [CoinGecko](https://www.coingecko.com/) website into [Deephaven](https://github.com/deephaven/deephaven-core).

## Variables
- **`timeToWatch`:** Integer number of minutes to run the script.
- **`secondsToSleep`:** Integer number of seconds between data pulls.  It is recommended that this number is 10 or higher. Note that too frequent requests will generate an HTTP status: '429 Too Many Requests'.
- **`getHistory`:** Boolean value.
   -  `false` for only live data.
   -  `true` for live and historical data.

> <<<<<<< HEAD

Reviewer: merge stuff

- **`daysHistory`:** Integer number of days to collect history. Data has automatic granularity.
   - Minutely data will be used for a duration within 1 day.
   - Hourly data will be used for a duration between 1 day and 90 days.
   - Daily data will be used for a duration above 90 days.
- **`ids`:** Array of coins used to pull information. For a full and current list of values, execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`
=======
- **`daysHistory`:** Integer number of days to collect history. Data has automatic granularity. 
   - Minutely data will be returned for a duration within 1 day.
   - Hourly data will be returned for a duration between 1 day and 90 days.
   - Daily data will be returned for a duration above 90 days.
- **`ids`:** Array of coins to pull information on. For a full and current list of values, execute the following: `curl -X GET "https://api.coingecko.com/api/v3/coins/list" -H "accept: application/json"`

> >>>>>>> 8dc4b156f20cd7726a71c15534164dd7407facb0

Reviewer: weird merge stuff
